src/log.js

```diff
@@ -224,46 +193,40 @@ export class Log {
    */
   defaultLevelWithFunc_() {
     // Delegate to the specific resolver.
-    return this.levelFunc_(parseInt(getMode().log, 10), getMode().development);
+    return this.levelFunc_(getMode().log, getMode().development);
   }
 
   /**
    * @param {string} tag
    * @param {!LogLevel} level
    * @param {!Array} messages
-   * @return {boolean} true if a message was logged
+   * @return {boolean} true if a the message was logged
    */
   msg_(tag, level, messages) {
-    if (this.getLevel_() < level) {
+    if (level > (levelOverride_ ?? this.level_)) {
       return false;
     }
-    let fn = this.win.console.log;
-    if (level == LogLevel.ERROR) {
-      fn = this.win.console.error || fn;
-    } else if (level == LogLevel.INFO) {
-      fn = this.win.console.info || fn;
-    } else if (level == LogLevel.WARN) {
-      fn = this.win.console.warn || fn;
-    }
+
+    const cs = this.win.console;
+    const fn =
```

Comment:
clever, but IMO harder to read. not b/c the pattern is inherently worse, its just super uncommon